src/MicrosoftTeams.ts

```diff
@@ -72,19 +74,34 @@ namespace microsoftTeams
         initializeCalled = true;
 
         // Undocumented field used to mock the window for unit tests
-        let currentWindow = this._window as Window || window;
+        currentWindow = this._window as Window || window;
+        let messageListener: (evt: MessageEvent) => void;
 
-        // If we are in an iframe then our parent window is the one hosting us (i.e. window.parent); otherwise,
-        // it's the window that opened us (i.e. window.opener)
+        // If we are in an iframe then our parent window is the one hosting us (i.e. window.parent)
         parentWindow = (currentWindow.parent !== currentWindow.self) ? currentWindow.parent : currentWindow.opener;
-        if (!parentWindow)
+        if (parentWindow)
         {
-            throw new Error("This page must be loaded in an iframe");
-        }
+            // Listen for messages post to our window (in a way that works for all browsers)
+            messageListener = (evt: MessageEvent) => processMessage(evt);
+            currentWindow.addEventListener("message", messageListener, false);
 
-        // Listen for messages post to our window (in a way that works for all browsers)
-        let messageListener = (evt: MessageEvent) => processMessage(evt);
-        currentWindow.addEventListener("message", messageListener, false);
+            try
+            {
+                // Send the initialized message to any origin since at this point we most likely don't know what our
+                // parent window's origin is yet and this message contains no data that could pose a security risk.
+                parentOrigin = "*";
+                let messageId = sendMessage("initialize");
+                callbacks[messageId] = (context: string, inDesktopClient: boolean) =>
```

Comment:
This should become version 0.3 of our SDK and we will only ask tabs to switch to it once the corresponding FE client changes have been made and deployed to PROD. The SDK will actually continue to work but desktop auth will be broken until the FE client changes have been implemented.